Rubberduck.Parsing/Grammar/VBAParser.g4

```diff
@@ -895,10 +895,16 @@ endOfLine :
     | whiteSpace? commentOrAnnotation
 ;
 
-// we expect endOfStatement to consume all trailing whitespace
+// We expect endOfStatement to consume all trailing whitespace.
+// We have to special case the end of file since infiniftly mant EOF tokens can be consumed at the end of file.
 endOfStatement :
-    (endOfLine whiteSpace? | (whiteSpace? COLON whiteSpace?))+
-    | whiteSpace? EOF
+    individualNonEOFEndOfStatement+ | whiteSpace? EOF
+;
+
+// we expect endOfStatement to consume all trailing whitespace
+individualNonEOFEndOfStatement :
```

Comment:
maybe name this `endOfInstruction`? `individualNonEOFEndOfStatement`  seems rather unwieldy :/

Reply:
I do not really like `endOfInstruction` because it is legal to end an instruction in whitespace followed by EOF. I just cannot include this here because one must never potatially match EOF in a loop, as explained in the comment to `endOfStatement`.